test/parallel/test-stream-construct-async-error.js

```diff
@@ -1,239 +0,0 @@
-    }
-  }
-
-  const foo = new Foo();
-  foo.end('hello');
-  foo.on('finish', common.mustCall());
-}
-
-{
-  class Foo extends Writable {
-    _write = common.mustCall((chunk, encoding, cb) => {
-      cb();
-    });
-
-    async _final(callback) {
-      // eslint-disable-next-line no-restricted-syntax
-      await setTimeout(common.platformTimeout(1));
-      callback();
-    }
-  }
-
-  const foo = new Foo();
-  foo.end('hello');
-  foo.on('finish', common.mustCall());
-}
-
-{
-  class Foo extends Writable {
-    _write = common.mustCall((chunk, encoding, cb) => {
-      cb();
-    });
-
-    async _final() {
-      // eslint-disable-next-line no-restricted-syntax
-      await setTimeout(common.platformTimeout(1));
-      throw new Error('boom');
-    }
-  }
-
-  const foo = new Foo();
-  foo.end('hello');
-  foo.on('error', common.expectsError({
-    message: 'boom'
-  }));
-  foo.on('close', common.mustCall());
-}
-
-{
-  const expected = ['hello', 'world'];
-  class Foo extends Transform {
-    async _flush() {
-      return 'world';
-    }
-
-    _transform(chunk, encoding, callback) {
-      callback(null, chunk);
-    }
-  }
-
-  const foo = new Foo();
-  foo.end('hello');
-  foo.on('data', common.mustCall((chunk) => {
-    assert.strictEqual(chunk.toString(), expected.shift());
-  }, 2));
-}
-
-{
-  const expected = ['hello', 'world'];
-  class Foo extends Transform {
-    async _flush(callback) {
-      callback(null, 'world');
-    }
-
-    _transform(chunk, encoding, callback) {
-      callback(null, chunk);
-    }
-  }
-
-  const foo = new Foo();
-  foo.end('hello');
-  foo.on('data', common.mustCall((chunk) => {
-    assert.strictEqual(chunk.toString(), expected.shift());
-  }, 2));
-}
-
-{
-  class Foo extends Transform {
-    async _flush(callback) {
-      throw new Error('boom');
-    }
-
-    _transform(chunk, encoding, callback) {
-      callback(null, chunk);
-    }
-  }
-
-  const foo = new Foo();
-  foo.end('hello');
-  foo.on('data', common.mustCall());
-  foo.on('error', common.expectsError({
-    message: 'boom'
-  }));
-  foo.on('close', common.mustCall());
-}
-
-{
-  class Foo extends Transform {
-    async _transform(chunk) {
-      return chunk.toString().toUpperCase();
-    }
-  }
-
-  const foo = new Foo();
-  foo.end('hello');
-  foo.on('data', common.mustCall((chunk) => {
-    assert.strictEqual(chunk.toString(), 'HELLO');
-  }));
-}
-
-{
-  class Foo extends Transform {
-    async _transform(chunk, _, callback) {
-      callback(null, chunk.toString().toUpperCase());
-    }
-  }
-
-  const foo = new Foo();
-  foo.end('hello');
-  foo.on('data', common.mustCall((chunk) => {
-    assert.strictEqual(chunk.toString(), 'HELLO');
-  }));
-}
-
-{
-  class Foo extends Transform {
-    async _transform() {
-      throw new Error('boom');
-    }
-  }
-
-  const foo = new Foo();
-  foo.end('hello');
-  foo.on('error', common.expectsError({
-    message: 'boom'
-  }));
-  foo.on('close', common.mustCall());
-}
```

Comment:
Shouldn't the tests be kept? I mean the these async versions should work mostly like their sync counterparts (with the exception of throwing code).

Or does it enforce some behaviour that should be unspecified?

Reply:
The async works totally differently.